src/Swarm/Game/State.hs

```diff
@@ -260,6 +261,32 @@ instance Monoid (Notifications a) where
 
 makeLenses ''Notifications
 
+data SolutionSource
+  = ScenarioSuggested
+  | -- | Includes the SHA1 of the program text
+    PlayerAuthored String
+
+data CodeToRun = CodeToRun SolutionSource ProcessedTerm
+
+getParsedInitialCode :: Maybe FilePath -> ExceptT Text IO (Maybe CodeToRun)
+getParsedInitialCode toRun = case toRun of
+  Nothing -> return Nothing
+  Just filepath -> do
+    contents <- liftIO $ TIO.readFile filepath
+    pt@(ProcessedTerm (Module (Syntax' srcLoc _ _) _) _ _) <-
```

Comment:
Why not use a `let` with lenses if you need that `srcLoc`?

Reply:
What would the lens expression look like?